### PR TITLE
chore: setup dev env without docker

### PR DESCRIPTION
**Pull request by IrakliJani**
### Description:

I managed to run lightdash without docker on my machine.
Motivation: app build time is much (MUCH) faster without docker.

**instructions to install on mac using brew/nvm:**
1. install [brew](https://brew.sh)
2. install node 16x using [nvm](https://github.com/nvm-sh/nvm)
    - `nvm install v16.17.0`
    - `nvm alias default v16.17.0`
3. install postgres
    - `brew install postgresql@14`
4. [install dbt](https://docs.getdbt.com/dbt-cli/install/homebrew)
    - `brew install dbt-labs/dbt/dbt-postgres@1.1.0`
5. before running yarn install ODBC library (required by [node-odbc](https://github.com/wankdanker/node-odbc))
    - `brew install unixODBC`
6. install packages
    - `yarn`
7. update `.env` and `docker/.env` according to your machine
8. source ENV files
    - TBD...
9. build / migrate / seed
    - `./docker/scripts/build.sh`
    - `./docker/scripts/seed-jaffle.sh`
    - `./docker/scripts/migrate.sh`
    - `./docker/scripts/seed-lightdash.sh`
10. run
    - `yarn dev`

I have few questions/obstacles and want to hear what @lightdash/engineering thinks first.

questions:
- what is the difference betwee an `.env` file and `docker/.env`
- I needed to override both of the env files in order the app  to work (otherwise they are only suitable for docker) how can we keep overrided .env files (they need to be `.gitignore`d) do you think having `.env.override`s is good?
- I feel like docker/scripts/* does not belong there and should be moved to the APP's root dir. thoughts?
- where should I document this after I finish this. docs section (somewhere under Installation + operation) or [CONTRIBUTING GUIDE](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#setup-development-environment), both?

### CODE REVIEW

**Comment by IrakliJani** [comment]
@lightdash/engineering

**Comment by ZeRego** [comment]
> what is the difference between an .env file and docker/.env

.env ( this was used in install.sh some time ago, I don't think we use this anymore 🤔 )
.env.fast-install ( is used in install.sh )
/docker/.env ( is used for local development env ) 

> I needed to override both of the env files in order the app to work (otherwise they are only suitable for docker) how can we keep overrided .env files (they need to be .gitignored) do you think having .env.overrides is good?

We can use one of the already ignored patterns
```
.env.local
.env.development.local
.env.test.local
.env.production.local
```

> I feel like docker/scripts/* does not belong there and should be moved to the APP's root dir. thoughts?

Still inside a 'scripts' folder ?  and move the `install.sh` there. 
I would also move all the docker files from the root to the `docker` folder.

> where should I document this after I finish this. docs section (somewhere under Installation + operation) or [CONTRIBUTING GUIDE](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#setup-development-environment), both?

I would add this to the `setup-development-environment` docs since it not going to be useful for anyone else beside contributors. Btw, we used to have instructions on how to run without docker but felt it was too confusing and harder to maintain since the docker container installs a bunch of stuff that is needed like python and dbt plugins, etc.

**Comment by IrakliJani** [comment]
@ZeRego cool. action items:
- [ ] `docker/.env` and merge with `.env`
- [ ] `.env.[ENV].local` after loading `.env` if any
- [ ] `docker/scripts` to `scripts`
- [ ] `install.sh` to `scripts` (probably name it `install-docker.sh` or `setup-docker.sh`
- [ ] docker related stuff to `docker/`

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -249,7 +239,7 @@ git submodule update --init --recursive
 Note: before the next step make sure your docker has 4GB of memory ( Docker -> settings -> resources ) you should be able to manipulate the values here.
 
 cd docker
-docker compose -p lightdash-app -f docker-compose.dev.yml --env-file .env up --detach --remove-orphans 
+docker compose -p lightdash-app -f docker-compose.dev.yml --env-file .env up --detach --remove-orphans
```
Comment (ZeRego) [inline .github/CONTRIBUTING.md:242]: if we are inside ./docker will the command work ? as .env no longer exists in that folder ?

File: package.json [hunk 2]
```diff
@@ -51,7 +52,8 @@
     "build": "run-s common-build warehouses-build frontend-build backend-build",
     "build-published-packages": "run-s common-build warehouses-build cli-build",
     "start": "run-p backend-start",
-    "prepare": "husky install"
+    "prepare": "husky install",
+    "load:env": "dotenv -c -e .env"
```
Comment (ZeRego) [inline package.json:56]: should this be using `.env.local` ?

File: packages/backend/src/database/seeds/development/01_initial_user.ts [hunk 3]
```diff
@@ -145,9 +145,19 @@ export async function seed(knex: Knex): Promise<void> {
         encryptedCredentials = enc.encrypt(JSON.stringify(creds));
     } else {
         // Dev mode
-        encryptedCredentials = Buffer.from(
-            'bd2b8a1f5c50106edc43aa886e9865c73b209f3afea54245dd4106f80d5318bab30789096d43f17310f664080a1d96adc230fd3093b9f5a85e1e975ed20c46c9928821f2de87d59b1ef0dbbae6fc332f11ffae7485e43e67e26efecc9ea9a984510b0e52032646bd925e4db0e45253332416c4bc332f16ad5269ffdd60d18d1cd178400153f7e64ca54fe408e059bd99ee79f8fcfdac066cd2a6d6a1ed15eca4ddfffa50254b1f0e2daa7615ccb1bffaeacf315641856eb4c0284012b5fc331a84ea3cbff7da85ed84e1005ab004b104d75b207723259055f83f62eae69eda1ccaeeecdf9d98226301d9c0a64c9313ec9cd127fd3dd11fc3925b8825b68a782870686c000cfc4703d8891b26',
-            'hex',
+        encryptedCredentials = enc.encrypt(
```
Comment (ZeRego) [inline packages/backend/src/database/seeds/development/01_initial_user.ts:148]: nice one 👍

File: packages/backend/src/index.ts [hunk 4]
```diff
@@ -210,7 +210,7 @@ app.use((error: Error, req: Request, res: Response, _: NextFunction) => {
 });
 
 // Run the server
-const port = process.env.PORT || 8080;
+const port = process.env.SERVER_PORT || 8080;
```
Comment (ZeRego) [inline packages/backend/src/index.ts:213]: Any reason to make this breaking change ?

Comment (IrakliJani) [reply packages/backend/src/index.ts:213]: I think it was breaking frontend. I'll double check and revert if it's not necesary

File: package.json (hunk 2 above)
Comment (IrakliJani) [reply package.json:56]: -c (cascading) flag does that

File: packages/backend/src/index.ts (hunk 4 above)
Comment (IrakliJani) [reply packages/backend/src/index.ts:213]: yeah, so webpack picks up PORT env variable and spawns its server with 8080. I will revert this once we resolve #3257

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (IrakliJani) [reply .github/CONTRIBUTING.md]: updated docks

**Review by ZeRego — COMMENTED** [review]
Everything is working well 🎉 well done 👍 

I only have a few change requests for the contributing docs:
- convert the bullet points into code blocks ( maybe everything to a single code block with comments )
- mention then need to fix the DBT_DEMO_DIR and LIGHTDASH_CONFIG_FILE path in the .env.local 
- documents on how to setup the postgres server ( even if is just linking to this article https://www.sqlshack.com/setting-up-a-postgresql-database-on-mac/ ) 

Screenshot 2022-09-30 at 16 56 00

**Comment by IrakliJani** [comment]
@ZeRego 
CleanShot 2022-10-04 at 15 04 59@2x

**Review by ZeRego — APPROVED** [review]

**Pull request merged**